packages/binding-coap/src/coap-server.ts

```diff
@@ -341,14 +371,17 @@ export default class CoapServer implements ProtocolServer {
                 if (segments.length === 2 || segments[2] === "") {
                     // Thing root -> send TD
                     if (req.method === "GET") {
-                        this.negotiateContentFormat(
+                        const negotiationSuccessful = this.negotiateContentFormat(
                             req,
                             res,
                             [ContentSerdes.DEFAULT, ContentSerdes.TD],
                             ContentSerdes.TD
                         );
-                        res.code = "2.05";
-                        res.end(JSON.stringify(thing.getThingDescription()));
+
+                        if (negotiationSuccessful) {
+                            res.code = "2.05";
+                            res.end(JSON.stringify(thing.getThingDescription()));
```

Comment:
Could the whole function could be simplified like this?

```JS
switch (accept) {
    case ContentSerdes.DEFAULT:
    case ContentSerdes.TD:
    case null:
        res.code = "2.05";
        res.setHeader("Content-Format", accept || ContentSerdes.TD);
        res.end(JSON.stringify(thing.getThingDescription()));
        break;
    default:
        res.code = "4.06";
        res.end(`Content-Format ${accept} is not supported by this resource.`);
        break;
}
```

Reply:
Hmm, yeah, I incorporated your suggestion in d1f3ae1965a9098051a345ac98682ff1d1ad6a35. My initial thoughts were that this method could be a generic one for content negotiation, but maybe that is a bit too much. Having a specific method could make more sense, at least for now.

Reply:
I think the generic solution would probably look like this:

```typescript
let serializer = SERIALIZERS[accept];
if (serializer) {
    let content = serializer.serialize(thing.getThingDescription());
    res.code = "2.05";
    res.setHeader("Content-Format", content.Type);
    res.end(content.Data);
} else {
    res.code = "4.06";
    res.end(`Content-Format ${accept} is not supported by this resource.`);
}
```

Reply:
The latest commits now implement a generic version using the `ContentSerdes` class for serialization. In the future, this would also allow for different TD representations (e.g., as CBOR or XML), but it seems as if these two codecs are currently not supported/implemented.